src/main/java/net/minecraftforge/common/EnumSimulate.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common;
+
+public enum EnumSimulate
```

Comment:
I thought we were trying to move away from smurf naming. How about `ActionSimulation`?

Reply:
`OperationType`/`ActionType`?

Reply:
I'm definitely open to better names, I had a hard time naming this and it's still not very good.

Reply:
I suggest spliting these methods into twos instead.

Reply:
I think `ActionType` sounds good, it's pretty short too.
Splitting methods into two isn't a great solution. It's best to share the code internally for simulate and execute, so splitting the public methods in two would just mean having the same code as this PR but with extra convenience methods publicly.